python/tank_vendor/shotgun_deploy/configuration.py

```diff
@@ -237,15 +238,36 @@ def get_tk_instance(self, sg_user):
 
         :param sg_user: Authenticated Shotgun user to associate
                         the tk instance with.
+
         """
-        # @todo - check if there is already a tank instance, in that case unload or warn
-        path = self._paths[sys.platform]
-        core_path = os.path.join(path, "install", "core", "python")
-        sys.path.insert(0, core_path)
-        import tank
+
+        core_path = os.path.join(self.path, "install", "core", "python")
+
+        # if the custom import handler has been added to `sys.meta_path`, then
+        # the import should work with the proper version of core. if not, then
+        # fallback to the behavior of manually adding this configuration's
+        # python dir to `sys.path` in order to pick it up.
+        try:
+            # custom import handler will handle this properly
+            import tank
```

Comment:
By the time this code runs, we expect that the custom import handler will already be installed on `sys.meta_path`. This `try/except` is just a fallback in case it hasn't been installed. Prior to this, the `core_path` was inserted prior to calling `import tank`, but thats not required for the custom importer to work.